### PR TITLE
Joe working

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
This PR includes the following updates:

- Staging models creation
- Staging macros creation
- CircleCi integration
- Issue and PR templates within the `.github/` folder
- Github docs addition

Still in progress:
- sqlfluff integration

### CODE REVIEW

File: models/stg_iterable.yml [hunk 1]
```diff
@@ -0,0 +1,419 @@
+version: 2
+
+models:
+  - name: stg_iterable__campaign_history
+    description: Table containing a history of campaigns.
+    loaded_at_field: _fivetran_synced
```
Comment (fivetran-jamie) [inline models/stg_iterable.yml:6]: i think `loaded_at_field` is just a `source` config right?

File: models/stg_iterable.yml [hunk 2]
```diff
@@ -0,0 +1,419 @@
...
+        description: User agent associated with the event
+      - name: user_agent_device
+        description: The device of the user agent
+  
+  - name: stg_iterable__event_extension
+    description: Table containing addition properties from the event table.
+    loaded_at_field: _fivetran_synced
+    columns:
+      - name: event_id
+        description: Unique identifier of the event. Generate by Fivetran
```
Comment (fivetran-jamie) [inline models/stg_iterable.yml:252]:
```suggestion
        description: Unique identifier of the event. Generated by Fivetran
```

File: models/stg_iterable__campaign_history.sql [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+fields as (
+
+    select
+        /*
+        The below macro is used to generate the correct SQL for package staging models. It takes a list of columns 
+        that are expected/needed (staging_columns from dbt_iterable_source/models/tmp/) and compares it with columns 
+        in the source (source_columns from dbt_iterable_source/macros/).
+        For more information refer to our dbt_fivetran_utils documentation (https://github.com/fivetran/dbt_fivetran_utils.git).
+        */
```
Comment (fivetran-jamie) [inline models/stg_iterable__campaign_history.sql:17]: this is great! i often wonder how confusing us using this macro is for users lol

File: models/stg_iterable__campaign_metrics.sql [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+        }}
+        
+    from base
+),
+
+final as (
+    
+    select 
+        id as campaign_id,
+        date,
```
Comment (fivetran-jamie) [inline models/stg_iterable__campaign_metrics.sql:33]: might wanna rename from `date` as i could see it causing issues

File: models/stg_iterable__event_extension.sql [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+            )
+        }}
+        
+    from base
+),
+
+final as (
+    select
+        _fivetran_id as event_id,
+        app_already_running,
```
Comment (fivetran-jamie) [inline models/stg_iterable__event_extension.sql:31]: some of the fields in this table are booleans right? should prefix with `_is` and stuff

File: macros/get_event_extension_columns.sql [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    {"name": "payload", "datatype": dbt_utils.type_string()},
+    {"name": "platform_endpoint", "datatype": dbt_utils.type_string()},
+    {"name": "push_message", "datatype": dbt_utils.type_string()},
+    {"name": "region", "datatype": dbt_utils.type_string()},
+    {"name": "sms_message", "datatype": dbt_utils.type_string()},
+    {"name": "sms_provider_response_code", "datatype": dbt_utils.type_int()},
+    {"name": "sms_provider_response_message", "datatype": dbt_utils.type_string()},
+    {"name": "sms_provider_response_more_info", "datatype": dbt_utils.type_string()},
+    {"name": "sms_provider_response_status", "datatype": dbt_utils.type_int()},
+    {"name": "sound", "datatype": dbt_utils.type_string()},
```
Comment (fivetran-jamie) [inline macros/get_event_extension_columns.sql:33]: the src yml said this was a boolean -- did it get cast as a string bc we didn't have source data?

File: models/stg_iterable__event_extension.sql [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+        badge,
+        canonical_url_id,
+        content_available,
```
Comment (fivetran-jamie) [inline models/stg_iterable__event_extension.sql:34]: boolean (i think)

**Review by fivetran-jamie — COMMENTED** [review]
left ya some comments!

File: .gitignore [hunk 8]
```diff
@@ -0,0 +1,4 @@
+
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply .gitignore:1]: Yes!

File: integration_tests/dbt_project.yml [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+name: 'iterable_source_integration_tests'
+version: '0.1.0'
+
+require-dbt-version: [">=0.18.0", "<0.20.0"]
+profile: 'integration_tests'
+
+vars:
+  iterable_source:
+    iterable_schema:                    iterable_source_integration_tests
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply integration_tests/dbt_project.yml:11]: Let's try removing and see

File: models/src_iterable.yml [hunk 10]
```diff
@@ -0,0 +1,380 @@
...
+      
+    freshness:
+      warn_after: {count: 72, period: hour}
+      error_after: {count: 168, period: hour}
+
+
+    tables:
+      - name: campaign_history
+        description: Table containing a history of campaigns.
+        loaded_at_field: _fivetran_synced
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_iterable.yml:19]: Nope, let's remove

File: models/src_iterable.yml [hunk 11]
```diff
@@ -0,0 +1,380 @@
...
+          - name: unique_pushes_delivered
+            description: The unique number of pushes delivered from the campaign
+          - name: unique_pushes_opened
+            description: The unique number of pushes opened from the campaign
+          - name: unique_pushes_sent
+            description: The unique number of pushes sent from the campaign
+          - name: unique_unsubscribes
+            description: The unique number of users who unsubscribed from your email
+
+      - name: campaign_supression_list_history #THIS IS SPELLED WRONG. PUT REQUEST IN TO CHANGE THE TYPO
```
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/src_iterable.yml:173]: That is a gooooooood question. Currently new customers will have the correct spelling and existing will have the old spelling.

Can we find a way to dynamically reference the two spellings?

File: models/stg_iterable.yml (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_iterable.yml:6]: Yup this was copied over we should delete

File: models/stg_iterable__campaign_metrics.sql (hunk 4 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_iterable__campaign_metrics.sql:33]: I agree

File: models/stg_iterable__event_extension.sql (hunk 5 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_iterable__event_extension.sql:31]: Yeah we should rename

File: macros/get_event_extension_columns.sql (hunk 6 above)
Comment (fivetran-joemarkiewicz) [reply macros/get_event_extension_columns.sql:33]: We should rerun these to ensure source data is accurate

File: models/stg_iterable__event_extension.sql (hunk 7 above)
Comment (fivetran-joemarkiewicz) [reply models/stg_iterable__event_extension.sql:34]: Same as above

**Comment by fivetran-jamie** [comment]
merged this into the `jamie-working` branch 🙂

**Pull request closed without merging**